lab_3/main.py

```diff
@@ -38,32 +71,58 @@ def _put_letter(self, letter: str) -> int:
         :param letter: a letter
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(letter, str):
+            return -1
+
+        id_number = 1
+        while True:
+            if letter in self.storage:
+                return 0
+            if id_number in self.storage.values():
+                id_number += 1
```

Comment:
Consider either defining a `id_number` field for the class or using some other way. This loop could be quite long.